src/resolve.rs

```diff
@@ -127,10 +130,16 @@ impl ResolveValue for ast::Expression {
                 select_default(variants).and_then(|variant| variant.value.to_value(env))
             }
             ast::Expression::AttributeExpression { ref id, ref name } => {
-                let attributes = env.ctx
-                    .get_message(&id.name)
-                    .as_ref()
-                    .and_then(|message| message.attributes.as_ref());
+                let attributes = if id.name.starts_with('-') {
+                    env.ctx
+                        .get_term(&id.name)
+                        .and_then(|message| message.attributes.as_ref())
+                } else {
+                    env.ctx
+                        .get_message(&id.name)
+                        .as_ref()
```

Comment:
I suspect that in general this file has a few `as_ref`s too many :) I think we can follow-up with a clean up later on, but in this particular case I'd like to see the same code in both branches of this `if-else` please.